bin/cylc-help

```diff
@@ -0,0 +1,524 @@
+                utility_commands,
+                hook_commands,
+                task_commands]:
+        for com, aliases in dct.items():
+            if any([alias == abbrev for alias in aliases]):
+                matches.clear()
+                matches.add(com)
+                finished_matching = True
+                break
+            if any([alias.startswith(abbrev) for alias in aliases]):
+                matches.add(com)
+        if finished_matching:
+            break
+    if not matches:
+        raise CommandNotFoundError('COMMAND not found: ' + abbrev)
+    if len(matches) > 1:
+        # multiple matches
+        raise CommandNotUniqueError('COMMAND "%s" not unique: %s' % (
+            abbrev,
+            ' '.join(['|'.join(all_commands[com]) for com in matches])))
+    return matches.pop()
+
+
+def pretty_print(incom, choose_dict, indent=True, numbered=False, sort=False):
+    # pretty print commands or topics from a dict:
+    # (com[item] = description)
+
+    if indent:
+        spacer = ' '
+    else:
+        spacer = ''
+
+    label = {}
+    choose = []
+    longest = 0
+    for item in choose_dict:
+        choose.append(item)
+        lbl = '|'.join(choose_dict[item])
+        label[item] = lbl
+        if len(lbl) > longest:
+            longest = len(lbl)
+
+    count = 0
+    pad = False
+    if len(choose) > 9:
+        pad = True
+
+    if sort:
+        choose.sort()
+    for item in choose:
+        if item not in incom:
+            raise SystemExit("ERROR: summary for '" + item + "' not found")
+
+        print spacer,
+        if numbered:
+            count += 1
+            if pad and count < 10:
+                digit = ' ' + str(count)
+            else:
+                digit = str(count)
+            print digit + '/',
+        print "%s %s %s" % (
+            label[item],
+            '.' * (longest - len(label[item])) + '...',
+            incom[item])
+
+
+def category_help(category):
+    coms = eval(category + '_commands')
+    alts = '|'.join(categories[category])
+    print 'CATEGORY: ' + alts + ' - ' + catsum[category]
+    print
+    print 'HELP: cylc [' + alts + '] COMMAND help,--help'
+    print '  You can abbreviate ' + alts + ' and COMMAND.'
+    print '  The category ' + alts + ' may be omitted.'
+    print
+    print 'COMMANDS:'
+    pretty_print(comsum, coms, sort=True)
+
+
+def typo(str):
+    corrected = str
+    if str == 'gcycl':
+        corrected = 'gcylc'
+    return corrected
+
+
+# BEGIN MAIN
+general_usage = """
+Cylc ("silk") is a suite engine and metascheduler that specializes in
+cycling weather and climate forecasting suites and related processing
+(but it can also be used for one-off workflows of non-cycling tasks).
+For detailed documentation see the Cylc User Guide (cylc doc --help).
+
+Version __CYLC_VERSION__
+
+The graphical user interface for cylc is "gcylc" (a.k.a. "cylc gui").
+
+USAGE:
+  % cylc -v,--version                   # print cylc version
```

Comment:
The short form `cylc -v` no longer works on this branch.

I suppose we should maintain back-compatibility here, even though `-v` is really standard (should really be paired with `--verbose`...).